ext/v8/property-callback-info.h

```diff
@@ -0,0 +1,104 @@
+// -*- mode: c++ -*-
+#ifndef RR_PROPERTY_CALLBACK_INFO_H
+#define RR_PROPERTY_CALLBACK_INFO_H
+
+namespace rr {
+
+  class PropertyCallbackInfo {
+  public:
+
+    template <class T>
+    class Base : public Wrapper<v8::PropertyCallbackInfo<T>> {
+    public:
+
+      inline Base(v8::PropertyCallbackInfo<T> info) :
+        Wrapper<v8::PropertyCallbackInfo<T>>(info) {}
+
+      inline Base(VALUE self) : Wrapper<v8::PropertyCallbackInfo<T>>(self) {}
+
+      static VALUE This(VALUE self) {
+        Base<T> info(self);
+        Locker lock(info->GetIsolate());
+        return Object(info->GetIsolate(), info->This());
+      }
+
+      static VALUE Data(VALUE self) {
+        Base<T> info(self);
+        Isolate isolate(info->GetIsolate());
+        Locker lock(isolate);
+
+        return PropertyCallback::unwrapData(isolate, info->Data()) ;
+      }
+
+      static VALUE GetIsolate(VALUE self) {
+        Base<T> info(self);
+        return Isolate(info->GetIsolate());
+      }
+
+    };
+
+    class Value : public Base<v8::Value> {
+    public:
+
+      inline Value(v8::PropertyCallbackInfo<v8::Value> info) :
+        Base<v8::Value>(info) {}
+
+      inline Value(VALUE self) : Base<v8::Value>(self) {}
+
+      static VALUE GetReturnValue(VALUE self) {
+        Value info(self);
+        Locker lock(info->GetIsolate());
+        return ReturnValue::Value(info->GetReturnValue());
```

Comment:
I think you can move this method into the template by saying `ReturnValue<T>` instead of `ReturnValue::Value`